Update Java dependency to >= 1.8

gkleiman opened the pull request:
Updated the deb/rpm Java dependencies, to require Java >= 1.8, as needed by Marathon v0.11.0.

Java 8 is available in Ubuntu >= 14.10, so we might have to stop supporting Ubuntu 12.04 (Precise) and Ubuntu 14.04 (Trusty), or document somewhere that Java 8 has to be installed from an unofficial repository.

aquamatthias commented:
@lingmann: I talked with Lukas a while ago about that topic. I can't remember the details, but the general outcome was: java 8 on our supported linux versions should not be a problem. Can you say something to ubuntu 12.04?

aquamatthias commented:
@gkleiman LGTM

gkleiman commented:
Ubuntu 14.04 LTS is also interesting, because it is the current LTS version.

lingmann commented:
Currently, our list of supported distros is discussed here:
https://mesosphere.com/blog/2014/07/17/mesosphere-package-repositories/

And I suspect that these changes will break building Marathon packages in Teamcity:
https://teamcity.mesosphere.io/viewType.html?buildTypeId=Oss_Marathon_SnapshotPackages

Before these changes can be merged, we will need to verify that the Marathon packages build for all of our supported distributions. I suspect it might break things on CentOS6/RHEL6 and perhaps some of the other distros we support. So to get this merged in, we'll need to test and develop an alternative java install solution for those distros. Also, documentation will need to be updated on the website, etc. etc.

gkleiman commented:
cc/ @aquamatthias 

Our TC agents have Java 8 installed and they have been using it to compile Marathon since we changed the SBT configuration weeks ago.

This commit only changes the dependencies of the native packages, so that they cannot be installed if Java 8 is not there.

We changed the list of the supported distros with the release of Mesos v0.23: https://mesosphere.com/blog/2015/07/09/deprecating-support-for-older-linux-distros-with-mesos-0-23/. We have to update the blog post you linked.

The following currently supported distros don't have an official package for Java 8:

- Ubuntu 12.04 (LTS)
- Ubuntu 14.04 (LTS)
- Debian 8 (Jessie)

We will have to include a reference to the Oracle Java 8 packages in our docs.

It is possible to install the java-1.8.0-openjdk package in CentOS 6, so no change should be needed for CentOS/RHEL.

aquamatthias commented:
@lingmann The packages can be build. TeamCity should be set up to build that.
The packages generated will have a dependency to java8. 
As mentioned by Gaston: Ubuntu 12.04 (LTS), Ubuntu 14.04 (LTS) and Debian 8 (Jessie) don't have java8 in the default repo. To use the packages there the java8 ppa has to be installed manually.
That means it is perfectly possible to use Marathon on those systems, but an additional step has to be taken. 

My suggestion:
- [ ] Merge this PR, build packages and try on those platforms.
- [ ] Describe how to install Marathon (and any other java based framework) on Ubuntu 12.04 (LTS), Ubuntu 14.04 (LTS) and Debian 8 (Jessie). Is that something the Marathon team should do?
- [ ] Part of the Release Notes will be a link to the updated installation guide.

aquamatthias commented:
For point 2 see https://github.com/mesosphere/marathon/issues/2214
